### ISSUE TITLE
Records' ToString can get minor performance improvements

### ISSUE THREAD

**Issue by AlFasGD** [dotnet/roslyn#54600]
# Summary

Using dotPeek I was able to find the way `ToString` works for a record of my choice, and realized that it could see some minor improvements. Specifically, two of them:
- Can call `StringBuilder.Append(char)`
- String literals are split and individually appended to the builder instead of being concatenated at compilation

# Example

![Example decompilation result](https://user-images.githubusercontent.com/8298332/124382303-5f4aa300-dccf-11eb-9c2f-a6c6b96fa94e.png)

In the provided example, the improvements are the following:
- `builder.Append('}');` in the second from last line in `ToString()`
- `builder.Append("Min = ");` in the first lines of `PrintMembers`
- `builder.Append(", Max = ");` in the last lines of `PrintMembers`

# Gains

To my knowledge, the gains from those improvements are fewer pointer dereferences, and one less loop execution in total. In real-world examples, this could barely see any beneficial changes.

**Comment by jcouv** [comment]
`ToString` on records is mostly designed to support a good debugging experience. That's why it doesn't include overload with formatting options for example.
I would recommend leaving the current implementation as-is.

**Comment by jaredpar** [comment]
> In real-world examples, this could barely see any beneficial changes.

What is the motivation for wanting to take this change then? This essentially adds complexity with little to no real beneficial changes. 

To take changes here I think we'd want to see concrete reasons or measurements that justified the added complexity to the implementation.

**Comment by RikkiGibson** [comment]
I agree that the motivation to make the change is low. I would still be interested to see a microbenchmark e.g. with [BenchmarkDotNet](https://benchmarkdotnet.org) before and after the proposed change, though.
